File: articles/decision-science-in-financial-risk-management/julia/capital_resilience_model.jl
# capital_resilience_model.jl

function capital_next(current_capital, period_return_pct, floor)
    return max(floor, current_capital * (1.0 + period_return_pct / 100.0))
end

println("Capital next = ", round(capital_next(100.0, -8.5, 20.0), digits=6))

File: articles/decision-science-in-financial-risk-management/python/capital_resilience_model.py
#!/usr/bin/env python3
"""Capital resilience helper."""

from __future__ import annotations


def capital_next(current_capital: float, period_return_pct: float, floor: float = 20.0) -> float:
    return max(floor, current_capital * (1.0 + period_return_pct / 100.0))


def capital_trigger_hit(capital: float, trigger: float = 72.0) -> bool:
    return capital < trigger


if __name__ == "__main__":
    c = capital_next(100.0, -8.5)
    print(round(c, 6))
    print(capital_trigger_hit(c))
